### PR TITLE
Updated auto-updater.md

### PR DESCRIPTION
**Pull request by Ajeey**
Missing documentation on the issue https://github.com/electron/electron/issues/10903

### CODE REVIEW

File: docs/api/auto-updater.md [hunk 1]
```diff
@@ -109,10 +109,12 @@ using this API.
 Restarts the app and installs the update after it has been downloaded. It
 should only be called after `update-downloaded` has been emitted.
 
-**Note:** `autoUpdater.quitAndInstall()` will close all application windows
+**Note 1:** `autoUpdater.quitAndInstall()` will close all application windows
 first and only emit `before-quit` event on `app` after that. This is different
 from the normal quit event sequence.
 
+**Note 2:** If you don't call `autoUpdater.quitAndInstall()` after `update-downloaded` event has been emitted then you will get the updated app on the next start/run of your app.
+
```
Comment (develar) [inline docs/api/auto-updater.md]: Not clear that actually update will be installed on quit. i.e. yes, result is that user will get updated app on next launch, but for developer maybe important to understand when update will be installed.

Comment (joshaber) [reply docs/api/auto-updater.md]: Speaking for Squirrel.Mac, the update is installed when the app quits. I _think_ that's also true of Squirrel.Windows, but I can't say for certain.

Comment (MarshallOfSound) [reply docs/api/auto-updater.md]: Squirrel.Windows installs while the app is still running, quitAndInstall and windows basically just restarts the app from the squirrel trampoline, the install is already done

File: docs/api/auto-updater.md [hunk 2]
```diff
@@ -109,10 +109,12 @@ using this API.
 Restarts the app and installs the update after it has been downloaded. It
 should only be called after `update-downloaded` has been emitted.
 
-**Note:** `autoUpdater.quitAndInstall()` will close all application windows
+**Note 1:** `autoUpdater.quitAndInstall()` will close all application windows
 first and only emit `before-quit` event on `app` after that. This is different
 from the normal quit event sequence.
```
Comment (zeke) [inline docs/api/auto-updater.md]: > This is different from the normal quit event sequence.

Can you extend this sentence to explain the difference?

I would also recommend changing the numbered notes to `**Note:**` for consistency with the rest of the docs

File: docs/api/auto-updater.md (hunk 1 above)
Comment (JerryGoyal) [reply docs/api/auto-updater.md]: unfortunately this doesn't work for background running apps. if I shutdown or restart the windows, on next bootup the app still holds the older version and the update cycle continues unless i explicitly "quit" the app.

**Comment by zeke** [comment]
@Ajeey are you able to make some changes based on the feedback above?

File: docs/api/auto-updater.md (hunk 2 above)
Comment (Ajeey) [reply docs/api/auto-updater.md]: @zeke 
> Can you extend this sentence to explain the difference?

The normal event sequence on quit is ```before-quit```, ```will-quit``` & ```quit```. 

Whereas, the quit sequence after calling ```autoUpdater.quitAndInstall()``` is ```window-all-closed```, ```before-quit```, ```will-quit``` & ```quit```.

Clear enough?

**Comment by Ajeey** [comment]
Can somebody help me understand why "continuous-integration/jenkins/pr-head" is failing?

**Comment by jkleinsc** [comment]
@Ajeey you can ignore the pr-head failure.  There was a setting on our CI that had to be changed.

File: docs/api/auto-updater.md (hunk 2 above)
Comment (zeke) [reply docs/api/auto-updater.md]: @Ajeey yep that sounds clear. 👍

Comment (ckerr) [reply docs/api/auto-updater.md]: That's much clearer. Could you extend this in the documentation too?

**Comment by zcbenz** [comment]
The failing test is not related to this PR, merging anyway.

**Pull request merged**